### PR TITLE
Adoption - hook post OSP 17.1 OC ironic

### PR DESCRIPTION
**Pull request by hjensas**
Add a hook to set up the OSP 17.1 overcloud ironic.

* ironic-python-agent glance images
* ironic provisioning network
* enroll ironic nodes
    
Jira: [OSPRH-12423](https://issues.redhat.com//browse/OSPRH-12423)

### CODE REVIEW

**Comment by pablintino** [comment]
@hjensas I guess this needs a rebase? As the change to include the ironic net is already merged

**Comment by hjensas** [comment]
> @hjensas I guess this needs a rebase? As the change to include the ironic net is already merged

Won't the merge bot/github just figure that out? I think we can just approve and lgtm and then it will figures itself out? (Not 100% sure, but I think so - doing so will save CI resources.)

**Comment by pablintino** [comment]
/approve

@hjensas let's try, but I'm not sure haha

**Comment by hjensas** [comment]
> /approve
> 
> @hjensas let's try, but I'm not sure haha

I asked in openstack-k8s-operators, the respons is "needs a rebase to fix the tree" :(
So ... I rebased it.

I want gerrit again ... :upside_down_face:

**Comment by EliadCohen** [comment]
@jparoly for awareness

File: hooks/playbooks/adoption_ironic_post_oc.yml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                  --container-format ari \
+                  --disk-format ari \
+                  --file ironic-python-agent.initramfs
+          args:
+            chdir: "{{ ansible_user_dir }}/ironic-python-agent"
+
+    - name: Create ironic provisioning network
+      delegate_to: osp-undercloud-0
+      environment:
+        OS_CLOUD: overcloud
```
Comment (ciecierski) [inline hooks/playbooks/adoption_ironic_post_oc.yml:77]: You have parameter for OS_CLOUD name defined in https://github.com/openstack-k8s-operators/data-plane-adoption/blob/main/scenarios/uni01alpha.yaml#L37 
`overcloud` should be replaced with `cifmw_adoption_osp_deploy_scenario.stacks[0].stackname`

File: hooks/playbooks/adoption_ironic_post_oc.yml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            path: "{{ ansible_user_dir }}/ci-framework-data/parameters"
+
+        - name: Write ironic_nodes.yaml on osp-unercloud-o
+          ansible.builtin.copy:
+            content: "{{ _ironic_nodes_slurp.content | b64decode }}"
+            dest: "{{ ansible_user_dir }}/ci-framework-data/parameters/ironic_nodes.yaml"
+
+        - name: Run baremetal create command to enroll the nodes in the Ironic service
+          environment:
+            OS_CLOUD: overcloud
```
Comment (ciecierski) [inline hooks/playbooks/adoption_ironic_post_oc.yml:120]: Same as for previous hard-coded OS_CLOUD

File: hooks/playbooks/adoption_ironic_post_oc.yml (hunk 1 above)
Comment (hjensas) [reply hooks/playbooks/adoption_ironic_post_oc.yml]: Done.

File: hooks/playbooks/adoption_ironic_post_oc.yml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      delegate_to: controller-0
+      register: _ironic_nodes_slurp
+      ansible.builtin.slurp:
+        src: "{{ ansible_user_dir }}/ci-framework-data/parameters/ironic_nodes.yaml"
+
+    - name: Enroll ironic nodes
+      delegate_to: osp-undercloud-0
+      block:
+        - name: Ensure directory exists
+          delegate_to: osp-undercloud-0
```
[the remark this answers is not in the document]
Comment (hjensas) [reply hooks/playbooks/adoption_ironic_post_oc.yml]: Done.

**Comment by ciecierski** [comment]
For anybody interested how the hook is connected to data-plane-adoption: https://github.com/openstack-k8s-operators/data-plane-adoption/pull/765

File: hooks/playbooks/adoption_ironic_post_oc.yml (hunk 2 above)
Comment (hjensas) [reply hooks/playbooks/adoption_ironic_post_oc.yml]: Done

File: hooks/playbooks/adoption_ironic_post_oc.yml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+          ansible.builtin.unarchive:
+            src: /usr/share/rhosp-director-images/ironic-python-agent-latest.tar
+            dest: "{{ ansible_user_dir }}/ironic-python-agent"
+            remote_src: true
+
+        - name: Upload ironic-python-agent kernel to glance
+          environment:
+            OS_CLOUD: "{{ cifmw_adoption_osp_deploy_scenario.stacks[0].stackname }}"
+          ansible.builtin.shell:
+            cmd: >-
```
[the remark this answers is not in the document]
Comment (ciecierski) [reply hooks/playbooks/adoption_ironic_post_oc.yml:50]: Discussed about ci_script in slack. You can ignore my comment about ci_script.

**Review by ciecierski — APPROVED** [review]
approved

**Comment by pablintino** [comment]
/lgtm
(replicating GH approval given by @ciecierski)

**Pull request merged**